tools/ctl/ide/explorer/explorer.go

```diff
@@ -316,6 +352,96 @@ func (e *Explorer) SelectTree() {
 	e.container.SwitchToPage(TreePageName)
 }
 
+func find(root *tview.TreeNode, nodeID string) (node *tview.TreeNode) {
+	children := root.GetChildren()
+	for _, child := range children {
+		data, ok := child.GetReference().(*nodeData)
+		if !ok {
+			continue
+		}
+		if data.NodeID == nodeID {
+			return child
+		}
+	}
+	return nil
+}
+
+func findOrAdd(root *tview.TreeNode, nodeID string, nodeFn func() *tview.TreeNode) (node *tview.TreeNode, added bool) {
+	target := find(root, nodeID)
+	if target != nil {
+		return target, false
+	}
+	children := root.GetChildren()
+	node = nodeFn()
+	for i, child := range children {
+		data, ok := child.GetReference().(*nodeData)
+		if !ok {
+			continue
+		}
+		if nodeID > data.NodeID {
+			children = slices.Insert(children, i, node)
+			root.SetChildren(children)
+			return node, true
+		}
+	}
+	children = append(children, node)
+	root.SetChildren(children)
+	return node, true
+}
+
+func (e *Explorer) handleRunUpdate(r *rundex.Run) {
+	e.runs[r.ID] = *r
+	rg, _ := findOrAdd(e.root, r.BenchmarkName, func() *tview.TreeNode {
+		return e.makeRunGroupNode(r.BenchmarkName, []string{r.ID})
+	})
+	findOrAdd(rg, r.ID, func() *tview.TreeNode {
+		return e.makeRunNode(r.ID)
+	})
+	// TODO: Update the run group's number of runs.
+}
+
+func (e *Explorer) handleRebuildUpdate(r *rundex.Rebuild) {
+	run, ok := e.runs[r.RunID]
+	if !ok {
+		log.Printf("rebuild update from unknown run %s", r.RunID)
+		return
+	}
+	rg := find(e.root, run.BenchmarkName)
+	if rg == nil {
+		log.Printf("Unable to find rungroup for  %s", run.BenchmarkName)
```

Comment:
missing return here and below at 415. I think this means you might want have this return an error and log it at a higher level